backend/danswer/bots/slack/handlers/handle_message.py

```diff
@@ -7,111 +8,107 @@
 from danswer.configs.constants import DOCUMENT_SETS
 from danswer.db.engine import get_sqlalchemy_engine
+from danswer.db.models import SlackBotConfig
 from danswer.direct_qa.answer_question import answer_qa_query
 from danswer.server.models import QAResponse
 from danswer.server.models import QuestionRequest
+from danswer.utils.logger import setup_logger
+
+logger_base = setup_logger()
 
 
 def handle_message(
-    msg: str,
-    channel: str,
-    message_ts_to_respond_to: str | None,
-    sender_id: str | None,
+    message_info: SlackMessageInfo,
+    channel_config: SlackBotConfig | None,
     client: WebClient,
-    logger: logging.Logger,
-    skip_filters: bool = False,
-    is_bot_msg: bool = False,
     num_retries: int = DANSWER_BOT_NUM_RETRIES,
     answer_generation_timeout: int = DANSWER_BOT_ANSWER_GENERATION_TIMEOUT,
     should_respond_with_error_msgs: bool = DANSWER_BOT_DISPLAY_ERROR_MSGS,
     disable_docs_only_answer: bool = DANSWER_BOT_DISABLE_DOCS_ONLY_ANSWER,
-    respond_every_channel: bool = DANSWER_BOT_RESPOND_EVERY_CHANNEL,
-) -> None:
-    engine = get_sqlalchemy_engine()
-    with Session(engine) as db_session:
-        channel_name = get_channel_name_from_id(client=client, channel_id=channel)
-        slack_bot_config = get_slack_bot_config_for_channel(
-            channel_name=channel_name, db_session=db_session
+) -> bool:
+    """Potentially respond to the user message, returns bool for failure
+    Question thrown out due to filters is not considered failure"""
+    msg = message_info.msg_content
+    channel = message_info.channel_to_respond
+    message_ts_to_respond_to = message_info.msg_to_respond
+    sender_id = message_info.sender
+    bipass_filters = message_info.bipass_filters
+    is_bot_msg = message_info.is_bot_msg
+
+    logger = cast(
+        logging.Logger,
+        _ChannelIdAdapter(logger_base, extra={SLACK_CHANNEL_ID: channel}),
+    )
+
+    document_set_names: list[str] | None = None
+    if channel_config and channel_config.persona:
+        document_set_names = [
+            document_set.name for document_set in channel_config.persona.document_sets
+        ]
+
+    reflexion = ENABLE_DANSWERBOT_REFLEXION
+
+    # List of user id to send message to, if None, send to everyone in channel
+    send_to: list[str] | None = None
+    respond_tag_only = False
+    respond_team_member_list = None
+    if channel_config and channel_config.channel_config:
+        channel_conf = channel_config.channel_config
+        if not bipass_filters and "answer_filters" in channel_conf:
+            reflexion = "well_answered_postfilter" in channel_conf["answer_filters"]
+
+            if (
+                "questionmark_prefilter" in channel_conf["answer_filters"]
+                and "?" not in msg
+            ):
+                logger.info(
+                    "Skipping message since it does not contain a question mark"
+                )
+                return False
```

Comment:
The return is now bool for if there's a failure (eg. should we tell user something has gone wrong). See if you agree with all these return True/False cases